codeexamples/SYCLBufferExampleExecutingCUDAKernels/StaticLibraryUse/src/main.cpp

```diff
@@ -0,0 +1,234 @@
+        {
+            StaticLibFnsCUDAKernel::HelloWorldFacade();
+        });
+    });
+}
+        
+// Function to enqueue a CUDA kernel to continue performing further additions
+// on an existing vector of ints to an existing SYCL queue.
+void VectorAdditionUsingCUDA( sycl::queue &vQ, const VectorInt_t &vVecA, const VectorInt_t &vVecB, VectorInt_t &vVecSumParallel )
+{
+    std::cout << "Hello, about to execute VectorAdditionUsingCUDA()\n";
+
+    const int nElements = vVecA.size();
+
+    // Create the range object for the vectors managed by the buffer.
+    const sycl::range< 1 > rngVecItems( nElements );
+
+    // Create buffers that hold the data shared between the host and the devices.
+    // The buffer destructor is responsible to copy the data back to host when it
+    // goes out of scope.
+    sycl::buffer bufA( vVecA.data(), rngVecItems );
+    sycl::buffer bufB( vVecB.data(), rngVecItems );
+    sycl::buffer bufSumParallel( vVecSumParallel.data(), rngVecItems );
+    
+    // Submit a command group to the queue by a C++ function that contains the
+    // data access permission and device computation (kernel).
+    vQ.submit( [&]( sycl::handler &vH ) 
+    { 
+        sycl::accessor accA( bufA, vH, sycl::read_only );
+        sycl::accessor accB( bufB, vH, sycl::read_only );
+        sycl::accessor accSum( bufSumParallel, vH, sycl::write_only );
+    
+        vH.host_task( [=]( const sycl::interop_handle &vIh ) 
+        {
+            // Error code to check return values for CUDA calls
+            cudaError_t err = cudaSuccess;
+
+            // Make compatible CUDA device pointers
+            const CUdeviceptr accA_ptr = vIh.get_native_mem< sycl::backend::ext_oneapi_cuda >( accA );
+            const CUdeviceptr accB_ptr = vIh.get_native_mem< sycl::backend::ext_oneapi_cuda >( accB );
+            CUdeviceptr accSum_ptr = vIh.get_native_mem< sycl::backend::ext_oneapi_cuda >( accSum );
+            const Data_t *rawHostPtrAccA = reinterpret_cast< const Data_t * >( accA_ptr );
+            const Data_t *rawHostPtrAccB = reinterpret_cast< const Data_t * >( accB_ptr );
+            Data_t *rawHostPtrAccSum = reinterpret_cast< Data_t * >( accSum_ptr );
+            
+            // CUDA number of threads in each thread block
+            const int threadsPerBlock = 1024;
+            // CUDA number of thread blocks in grid
+            const int blocksPerGrid = (nElements + threadsPerBlock - 1) / threadsPerBlock;
+            // Call the CUDA kernel directly from SYCL. Pointers are to device data.
+            err = StaticLibFnsCUDAKernel::KernelCUDAVectorAddFacade( blocksPerGrid, threadsPerBlock, 
+                rawHostPtrAccA, rawHostPtrAccB, rawHostPtrAccSum, vVecA.size() );
+            if( err != cudaSuccess )  
+            { 
+                fprintf( stderr, "Failed to launch KernelCUDAVectorAddFacade (error code %s)!\n", cudaGetErrorString( err ) );
```

Comment:
Should this use `std::cerr` since the rest of the code is using `cout`?

Reply:
Code changed to use std::cerr. Done.